closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1044 @@
+            System.setProperty("javax.net.ssl.keyStore", keyStore);
+        }
+    }
+
+    /**
+     * Check the restricted security properties
+     */
+    private static void restrictsCheck() {
+        // Check restricts object
+        if (restricts == null) {
+            printStackTraceAndExit("Restricted security property is null.");
+        }
+
+        // Check if the SunsetDate expired.
+        if (isPolicySunset(restricts.descSunsetDate)) {
+            printStackTraceAndExit("Restricted security policy expired.");
+        }
+
+        // Check secure random settings.
+        if (isNullOrBlank(restricts.jdkSecureRandomProvider)
+                || isNullOrBlank(restricts.jdkSecureRandomAlgorithm)) {
+            printStackTraceAndExit("Restricted security mode secure random is null.");
+        }
+    }
+
+    /**
+     * Check if restricted security policy sunset.
+     *
+     * @param descSunsetDate the sun set date from java.security
+     * @return true if the restricted security policy sunset
+     */
+    private static boolean isPolicySunset(String descSunsetDate) {
+
+        boolean isSunset = false;
+        try {
+            isSunset = LocalDate.parse(descSunsetDate, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                    .isBefore(LocalDate.now());
+        } catch (DateTimeParseException except) {
+            printStackTraceAndExit(
+                    "Restricted security policy sunset date is incorrect, the correct format is yyyy-MM-dd.");
+        }
+
+        if (debug != null) {
+            debug.println("Restricted security policy is sunset: " + isSunset);
+        }
+        return isSunset;
+    }
+
+    /**
+     * Check if the input string is null or blank.
+     *
+     * @param string the input string
+     * @return true if the input string is null or blank
+     */
+    private static boolean isNullOrBlank(String string) {
+        return (string == null) || string.isBlank();
+    }
+
+    private static void printStackTraceAndExit(Exception exception) {
+        exception.printStackTrace();
+        System.exit(1);
+    }
+
+    private static void printStackTraceAndExit(String message) {
+        printStackTraceAndExit(new RuntimeException(message));
+    }
+
+    /**
+     * Nested class for the properties of restricted security
+     */
+    private static final class RestrictedSecurityProperties {
+
+        private String descName;
+        private String descNumber;
+        private String descPolicy;
+        private String descSunsetDate;
+
+        // Security properties.
+        private String jdkTlsDisabledNamedCurves;
+        private String jdkTlsDisabledAlgorithms;
+        private String jdkTlsDphemeralDHKeySize;
+        private String jdkTlsLegacyAlgorithms;
+        private String jdkCertpathDisabledAlgorithms;
+        private String jdkSecurityLegacyAlgorithm;
+        private String keyStoreType;
+        private String keyStore;
+
+        // For Secure Random.
```

Comment:
`SecureRandom` is the class name (no space), unless you are referring to the configuration of source of secure random numbers (no capitals).